llvm/lib/Support/RISCVISAInfo.cpp

```diff
@@ -854,6 +854,81 @@ RISCVISAInfo::parseArchString(StringRef Arch, bool EnableExperimentalExtension,
                              "string must be lowercase");
   }
 
+  bool IsProfile = Arch.starts_with("rvi") || Arch.starts_with("rva") ||
+                   Arch.starts_with("rvb") || Arch.starts_with("rvm");
+  std::string NewArch;
+  if (IsProfile) {
+    // A mapping from profile name to march string with all mandatory
+    // extensions.
+    static const std::map<StringLiteral, StringLiteral> SupportedProfiles = {
```

Comment:
This may not be fully evaluated at compile time.  I'd go back to the static arrays personally.

Reply:
Yeah this is going to be a global constructor which is against coding standards. https://llvm.org/docs/CodingStandards.html#do-not-use-static-constructors

Reply:
Is there a global constructor/destructor if it is inside a function and lazily initialized?

Reply:
I guess it will only construct on the first call so it isn't a global constructor. Sorry.

Reply:
I go back to use a global static array of RISCVProfile.